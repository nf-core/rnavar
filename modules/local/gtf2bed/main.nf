process GTF2BED {
    tag "$gtf"
    label 'process_low'

    conda "conda-forge::r-base=3.5.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-base:3.5.0' :
        'biocontainers/r-base:3.5.0'}"

    input:
    tuple val(meta), path(gtf)
    val feature_type

    output:
    path '*.bed'       , emit: bed
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script: // This script is bundled with the pipeline, in nf-core/rnaseq/bin/
    def allowed_type = ["exon", "transcript", "gene"];
    if (feature_type){
        feature_type = allowed_type.contains(feature_type) ? feature_type : "exon"
    }
    """
    Rscript --no-save -<<'RCODE'
        gtf = read.table("${gtf}", sep="\t")
        gtf = subset(gtf, V3 == "${feature_type}")
        write.table(data.frame(chrom=gtf[,'V1'], start=gtf[,'V4'], end=gtf[,'V5']), "tmp.exome.bed", quote = F, sep="\t", col.names = F, row.names = F)
    RCODE

    awk '{print \$1 "\t" (\$2 - 1) "\t" \$3}' tmp.exome.bed > exome.bed
    rm -rf tmp.exome.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Rscript: \$(echo \$(Rscript --version 2>&1) | sed 's/R scripting front-end version //')
    END_VERSIONS
    """
}
