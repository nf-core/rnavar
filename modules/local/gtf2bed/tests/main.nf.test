nextflow_process {

    name "Test Process GTF2BED"
    script "../main.nf"
    process "GTF2BED"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.value([
                    [id:'gtf'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)
                ])
                input[1] = "exon"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.bed,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }

    test("Should run without failures - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.value([
                    [id:'gtf'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)
                ])
                input[1] = "exon"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.bed,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }
}
