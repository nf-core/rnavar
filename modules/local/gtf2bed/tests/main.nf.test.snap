{
    "Should run without failures - stub": {
        "content": [
            [
                [
                    {
                        "id": "gtf"
                    },
                    "exome.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            {
                "GTF2BED": {
                    "Rscript": "3.5.1 (2018-07-02)"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-30T14:33:17.7904128"
    },
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "gtf"
                    },
                    "exome.bed:md5,17192418c24c524881e7d970f94a4a26"
                ]
            ],
            {
                "GTF2BED": {
                    "Rscript": "3.5.1 (2018-07-02)"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-30T14:33:04.885526129"
    }
}